### PR TITLE
 [PyTest #33 - Extract]  PyTest CI Pipelines

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
See title

### CODE REVIEW

File: .ci/kitchen-archlts-py3-pytest [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+runTestSuite(
+    concurrent_builds: 2,
+    distro_name: 'arch',
+    distro_version: 'lts',
+    env: env,
+    golden_images_branch: 'master',
+    jenkins_slave_label: 'kitchen-slave',
+    nox_env_name: 'pytest-zeromq',
+    nox_passthrough_opts: '',
```
Comment (bryceml) [inline .ci/kitchen-archlts-py3-pytest:11]: no passthrough opts?  are we testing everything now instead of just pkg tests?

Comment (s0undt3ch) [reply .ci/kitchen-archlts-py3-pytest:11]: We are. As part of the upcoming 3.8 support.

File: .ci/kitchen-centos7-py3-pycryptodome-pytest [hunk 2]
```diff
@@ -0,0 +1,16 @@
+@Library('salt@master-1.7') _
+
+runTestSuite(
+    concurrent_builds: 1,
```
Comment (bryceml) [inline .ci/kitchen-centos7-py3-pycryptodome-pytest:4]: why is this one 1 when the rest are 2 ?

Comment (s0undt3ch) [reply .ci/kitchen-centos7-py3-pycryptodome-pytest:4]: Dam, forgot to update it

File: .ci/kitchen-macosxhighsierra-py3-pytest [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// Pre-nox pipeline
+runTestSuite(
+    concurrent_builds: 2,
+    distro_name: 'macosx',
+    distro_version: 'highsierra',
+    env: env,
+    golden_images_branch: 'master',
+    jenkins_slave_label: 'kitchen-slave-mac',
+    kitchen_platforms_file: '/var/jenkins/workspace/pre-golden-nox-platforms.yml',
+    kitchen_verifier_file: '/var/jenkins/workspace/nox-verifier.yml',
```
Comment (bryceml) [inline .ci/kitchen-macosxhighsierra-py3-pytest:12]: this is the default so we don't have to define it right?

File: .ci/kitchen-macosxmojave-py3-pytest [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// Pre-nox pipeline
+runTestSuite(
+    concurrent_builds: 2,
+    distro_name: 'macosx',
+    distro_version: 'mojave',
+    env: env,
+    golden_images_branch: 'master',
+    jenkins_slave_label: 'kitchen-slave-mac',
+    kitchen_platforms_file: '/var/jenkins/workspace/pre-golden-nox-platforms.yml',
+    kitchen_verifier_file: '/var/jenkins/workspace/nox-verifier.yml',
```
Comment (bryceml) [inline .ci/kitchen-macosxmojave-py3-pytest:12]: same here

File: .ci/kitchen-macosxsierra-py3-pytest [hunk 5]
```diff
@@ -0,0 +1,22 @@
+@Library('salt@master-1.7') _
```
Comment (bryceml) [inline .ci/kitchen-macosxsierra-py3-pytest:1]: we don't support macos sierra anymore, we should delete this file.

File: .ci/kitchen-macosxhighsierra-py3-pytest (hunk 3 above)
Comment (s0undt3ch) [reply .ci/kitchen-macosxhighsierra-py3-pytest:12]: Hmm, yes?

File: .ci/kitchen-macosxsierra-py3-pytest (hunk 5 above)
Comment (bryceml) [reply .ci/kitchen-macosxsierra-py3-pytest:1]: we could maybe add catalina if we want.

File: .ci/kitchen-archlts-py3-pytest (hunk 1 above)
Comment (bryceml) [reply .ci/kitchen-archlts-py3-pytest:11]: should we add on --ssh-tests then ?

Comment (s0undt3ch) [reply .ci/kitchen-archlts-py3-pytest:11]: Sure. I can't update it today though.
Want to push that change in?

Comment (bryceml) [reply .ci/kitchen-archlts-py3-pytest:11]: sure

**Comment by bryceml** [comment]
re-run all full

**Review by bryceml — DISMISSED** [review]

**Review by bryceml — APPROVED** [review]
👍

File: .ci/kitchen-macosxhighsierra-py3 [hunk 6]
```diff
@@ -9,7 +9,6 @@ runTestSuite(
     golden_images_branch: 'master',
     jenkins_slave_label: 'kitchen-slave-mac',
     kitchen_platforms_file: '/var/jenkins/workspace/pre-golden-nox-platforms.yml',
-    kitchen_verifier_file: '/var/jenkins/workspace/nox-verifier.yml',
```
Comment (Ch3LL) [inline .ci/kitchen-macosxhighsierra-py3:12]: are we using a different verifier file? Im sure this is just my misunderstanding with the pytest changes?

Comment (s0undt3ch) [reply .ci/kitchen-macosxhighsierra-py3:12]: That's actually the default value, hence removing it.

**Review by Ch3LL — APPROVED** [review]

**Review by dwoz — APPROVED** [review]

**Pull request merged**